addons/mail/models/mail_alias.py

```diff
@@ -185,70 +216,63 @@ def _clean_and_check_mail_catchall_allowed_domains(self, value):
-                if icp_value and sanitized_name == icp_value:
-                    matching_alias_name = f'{sanitized_name}@{alias_domain}' if alias_domain else sanitized_name
-                    raise UserError(
-                        _('The e-mail alias %(matching_alias_name)s is already used as %(alias_duplicate)s alias. Please choose another alias.',
-                          matching_alias_name=matching_alias_name,
-                          alias_duplicate=icp_label_by_key[icp_key])
-                    )
+        if len(alias_names) != len(alias_domains):
+            msg = (f"Invalid call to '_check_unique': names and domains should make coherent lists, "
+                   f"received {', '.join(alias_names)} and {', '.join(alias_domain.name for alias_domain in alias_domains)}")
+            raise ValueError(msg)
+
+        # reorder per alias domain, keep only not void alias names (void domain also checks uniqueness)
+        domain_to_names = defaultdict(list)
+        for alias_name, alias_domain in zip(alias_names, alias_domains):
+            if alias_name and alias_name in domain_to_names[alias_domain]:
+                raise UserError(
+                    _('Email aliases %(alias_name)s cannot be used on several records at the same time. Please update records one by one.',
+                      alias_name=alias_name)
+                )
+            if alias_name:
+                domain_to_names[alias_domain].append(alias_name)
 
         # matches existing alias
-        matching_alias = self.env['mail.alias']
-        if valid_names:
-            domain = [('alias_name', 'in', valid_names)]
-            if self:
-                domain += [('id', 'not in', self.ids)]
-            matching_alias = self.search(domain, limit=1)
-        if not matching_alias:
-            return
-        if matching_alias.alias_parent_model_id and matching_alias.alias_parent_thread_id:
-            # If parent model and parent thread ID both are set, display document name also in the warning
-            document_name = self.env[matching_alias.alias_parent_model_id.model].sudo().browse(matching_alias.alias_parent_thread_id).display_name
-            raise UserError(
-                _('The e-mail alias %(matching_alias_name)s is already used by the %(document_name)s %(model_name)s. Choose another alias or change it on the other document.',
-                  matching_alias_name=matching_alias.display_name,
-                  document_name=document_name,
-                  model_name=matching_alias.alias_parent_model_id.name)
+        domain = expression.OR([
+            ['&', ('alias_name', 'in', alias_names), ('alias_domain_id', '=', alias_domain.id)]
+            for alias_domain, alias_names in domain_to_names.items()
+        ])
+        if domain and self:
+            domain += [('id', 'not in', self.ids)]
+        existing = self.search(domain, limit=1) if domain else self.env['mail.alias']
+        if existing:
```

Comment:
to save one indetation bellow
```suggestion
        if not existing:
            return